comp/core/workloadmeta/collectors/internal/process/process_collector_test.go

```diff
@@ -56,12 +60,15 @@ func setUpCollectorTest(t *testing.T, configOverrides map[string]interface{}) co
 
 	mockClock := clock.NewMock()
 	mockProbe := mocks.NewProbe(t)
-	processCollector := newProcessCollector(collectorID, workloadmeta.NodeAgent, mockClock, mockProbe)
 
-	return collectorTest{&processCollector, mockProbe, mockClock, mockStore, mockContainerProvider}
+	// mock language detection system probe config
+	mockSystemProbeConfig := configmock.New(t)
+	processCollector := newProcessCollector(collectorID, workloadmeta.NodeAgent, mockClock, mockProbe, mockSystemProbeConfig)
+
+	return collectorTest{&processCollector, mockProbe, mockSystemProbeConfig, mockClock, mockStore, mockContainerProvider}
 }
 
-func createTestProcess1(pid int32, createTime int64) *procutil.Process {
+func createTestPythonProcess1(pid int32, createTime int64) *procutil.Process {
```

Comment:
Nit: we don't need append the `1` here since the language is already a differentiator. 
```suggestion
func createTestPythonProcess(pid int32, createTime int64) *procutil.Process {
```